test/LiquidDemocracy.test.js

```diff
@@ -10,53 +10,100 @@ const should = require('chai')
+    describe('voting', function () {
+        it('should allow registered voters to vote', async function () {
+            await this.liquidDemocracyBallot.vote(accounts[0], 0);
+            await this.liquidDemocracyBallot.vote(accounts[1], 0);
+            await this.liquidDemocracyBallot.vote(accounts[2], 1);
+            const voteCountOption0 = await this.liquidDemocracyBallot.getBallotVoteCount(0);
+            const voteCountOption1 = await this.liquidDemocracyBallot.getBallotVoteCount(1);
+
+            voteCountOption0.toNumber().should.equal(2);
+            voteCountOption1.toNumber().should.equal(1);
+        });
+
+        it('should not allow unregistered voters to vote', async function () {
+            await assertRevert(this.liquidDemocracyBallot.vote(accounts[4], 0));
+        });
+
+        it('should not allow registered voters to vote more than once', async function () {
+            await this.liquidDemocracyBallot.vote(accounts[0], 0);
+            await assertRevert(this.liquidDemocracyBallot.vote(accounts[0], 0));
+        });
     });
-  });
 
-  describe('delegating', function () {
-    // TODO
-  });
+    describe('delegations produce the expected output', function () {
+
+        it('should perform a simple delegation', async function () {
+            await this.liquidDemocracyBallot.delegate(accounts[1], { from: accounts[0] });
+
+            const representative = await this.liquidDemocracyBallot.getMyRepresentative({ from: accounts[0] })
+            const representativeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[1] })
+            const representeeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[0] })
+
+            assert.equal(representative, accounts[1]);
+            assert.equal(representativeWeight, 2);
+            assert.equal(representeeWeight, 0);
+        });
+
+        it('should perform 1 hop transitive delegation', async function () {
+            await this.liquidDemocracyBallot.delegate(accounts[1], { from: accounts[0] });
+            await this.liquidDemocracyBallot.delegate(accounts[2], { from: accounts[1] });
+
+            const representative1 = await this.liquidDemocracyBallot.getMyRepresentative({ from: accounts[0] })
+            const representative2 = await this.liquidDemocracyBallot.getMyRepresentative({ from: accounts[1] })
+            const representative1Weight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[1] })
+            const representative2Weight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[2] })
+            const representeeWeight = await this.liquidDemocracyBallot.getMyWeight({ from: accounts[0] })
+
+            assert.equal(representative1, accounts[1]);
+            assert.equal(representative2, accounts[2]);
+            assert.equal(representative1Weight, 0);
+            assert.equal(representative2Weight, 3);
```

Comment:
Transitive delegation carries all weight? I think this is good to start simple. Let's [update the spec](https://github.com/DemocracyEarth/contracts/issues/13#issuecomment-380251717) though, right now it says '_A voter delegates exactly 1 vote_' and it's confusing in this transitive case. 

Also wondering if 'transitive' is the best word to use for this case, it makes more sense [in spanish](http://dle.rae.es/?id=aKeWba1) than english I think. Another option is talking about 'chained delegations'

Reply:
I added C.7 to the spec and modified C.5